tests/testutils.py

```diff
@@ -121,19 +121,35 @@ def get_credential(self):
         return self._g_credential
 
 
-class MockAdapter(adapters.HTTPAdapter):
-    """A mock HTTP adapter for the Python requests module."""
-    def __init__(self, data, status, recorder):
+class MockMultiRequestAdapter(adapters.HTTPAdapter):
+    """A mock HTTP adapter that supports multiple responses for the Python requests module."""
+    def __init__(self, responses, statuses, recorder):
+        """Constructs a MockMultiRequestAdapter.
+
+        The lengths of the responses and statuses parameters must match.
+
+        Each incoming request consumes a response and a status, in order. If all responses and
+        statuses are exhausted, further requests will reuse the last response and status.
+        """
         adapters.HTTPAdapter.__init__(self)
-        self._data = data
-        self._status = status
+        assert len(responses) == len(statuses)
```

Comment:
Throw a ValueError here. We only do asserts inside test cases.

Reply:
Done.